src/hooks/useVoucher/useVoucher.test.tsx

```diff
@@ -0,0 +1,256 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { useVoucher } from "./useVoucher";
+import { wait } from "@testing-library/react-native";
+import { postTransaction } from "../../services/quota";
+
+jest.mock("../../services/quota");
+const mockPostTransaction = postTransaction as jest.Mock;
+
+const key = "KEY";
+const endpoint = "https://myendpoint.com";
+
+const transactionTime = new Date(2020, 3, 1);
+
+const transactions = [
+  {
+    category: "voucher",
+    quantity: 2,
+    identifiers: [
+      {
+        label: "Merchant Code",
+        value: "CDC-0001"
+      },
+      {
+        label: "Redeemed by",
+        value: "Vendor"
+      }
+    ]
+  }
+];
+
+const mockPostTransactionResult = {
+  transactions: [
+    {
+      transaction: transactions,
+      transactionTime
+    }
+  ]
+};
+
+describe("useVoucher", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("update vouchers when removing voucher", () => {
+    it("should update vouchers when voucher is removed", async () => {
+      expect.assertions(1);
+      const { result } = renderHook(() => useVoucher(key, endpoint));
+
+      await wait(() => {
+        result.current.addVoucher({ serial: "123456789", denomination: 2 });
+      });
+
+      await wait(() => {
+        result.current.removeVoucher("123456789");
+      });
+
+      expect(result.current.vouchers).toStrictEqual([]);
+    });
```

Comment:
Should check if it actually got added, if not `addVoucher` and `removeVoucher` could do nothing and this test would still pass